internal/util/version.go

```diff
@@ -10,48 +10,91 @@ import (
 const (
 	UNKNOWN_VERSION  string = "???"
 	UNKNOWN_HASH     string = "????????"
-	VERSION_FILENAME string = "VERSION.txt"
+	VERSION_FILENAME string = "VERSION.json"
 	DIRTY_SUFFIX     string = "dirty"
 	HASH_LENGTH      int    = 8
 )
 
+type Version struct{
+	Short string `json:"short-version"`
+	Hash string `json:"git-hash"`
+	Status string `json:"status"`
+}
+
 func GetAutograderVersion() string {
 	versionPath := ShouldAbs(filepath.Join(ShouldGetThisDir(), "..", "..", VERSION_FILENAME))
 	if !IsFile(versionPath) {
 		log.Error("Version file does not exist.", log.NewAttr("path", versionPath))
 		return UNKNOWN_VERSION
 	}
 
-	version, err := ReadFile(versionPath)
+	var readVersion Version
+	readVersion.Short = UNKNOWN_VERSION
+
+	err := JSONFromFile(versionPath,&readVersion)
```

Comment:
I don't like how the file is read in two locations.
Just have one function that handles the file (computes path, logs, reads, etc) and return a `(*Version, nil)`.